src/scifem/solvers.py

```diff
@@ -0,0 +1,220 @@
+import dolfinx
+from typing import Callable
+import ufl
+import numpy as np
+from petsc4py import PETSc
+
+__all__ = ["NewtonSolver"]
+
+
+class NewtonSolver:
+    max_iterations: int
+    _bcs: list[dolfinx.fem.DirichletBC]
+    A: PETSc.Mat
+    b: PETSc.Vec
+    _J: dolfinx.fem.Form | ufl.form.Form
```

Comment:
`dolfinx.fem.Form` is the class. `dolfinx.fem.form` is a constructor for the class.
See: 
https://docs.fenicsproject.org/dolfinx/v0.8.0/python/generated/dolfinx.fem.html#dolfinx.fem.Form
https://docs.fenicsproject.org/dolfinx/v0.8.0/python/generated/dolfinx.fem.html#dolfinx.fem.form

Reply:
I think using the `dolfinx.fem.Form` in the type annotation is fine, because it represents an actual `type`. AFAICS `dolfinx.fem.form` is just a function that creates a `dolfinx.fem.Form` object. Therefore, it is not a type and is not valid in a type annotation.

Could this be?

Reply:
Yes.